torch/csrc/jit/codegen/cuda/arith.cpp

```diff
@@ -212,14 +347,14 @@ Val* reductionOp(
     uint_axes.push_back((unsigned int)axis);
   }
 
-  Val* out = tv->newForReduction(uint_axes);
+  TensorView* out = tv->newForReduction(uint_axes);
   if (init->getDataType().value() != v1->getDataType().value())
     init = castOp(v1->getDataType().value(), init);
   new ReductionOp(reduction_op_type, init, out, v1);
   return out;
 }
 
-TORCH_CUDA_API Val* sum(Val* v1, const std::vector<int>& axes) {
+TORCH_CUDA_API TensorView* sum(Val* v1, const std::vector<int>& axes) {
```

Comment:
We can hardcode the input here to be a TensorView, we should never call a reduction on anything else.

Reply:
Done at 5542fdd805bdca0fcb685cf8a3e33dbdf8b3f9d2. Also updated `reductionOp` as its `v1` parameter is also assumed to be `TensorView*`.